src/screens/NetworkSettings/CustomNetworkSettingsScreen.js

```diff
@@ -0,0 +1,260 @@
+};
+
+/**
+ * Check if the network settings status is loading.
+ * @param {object} networkSettingsStatus - status from redux store
+ * @returns {boolean} - true if the status is loading, false otherwise
+ */
+const isLoading = (networkSettingsStatus) => {
+  return networkSettingsStatus === NETWORKSETTINGS_STATUS.LOADING;
+};
+
+/**
+ * Verifies if the errorModel of the form has an error message. 
+ */
+function hasError(errorModel) {
+  return Object
+    .values({ ...errorModel })
+    .reduce((hasError, currValue) => hasError || !isEmpty(currValue), false);
+};
+
+/**
+ * Validates the formModel, returning the errorModel.
+ * If there is no error in the formModel, the errorModel is returned empty.
+ */
+function validate(formModel) {
+  const errorModel = {};
+
+  if (!formModel.nodeUrl) {
+    errorModel.nodeUrl = t`nodeUrl is required.`;
+  }
+
+  if (!formModel.explorerUrl) {
+    errorModel.explorerUrl = t`explorerUrl is required.`;
+  }
+
+  if (!formModel.explorerServiceUrl) {
+    errorModel.explorerServiceUrl = t`explorerServiceUrl is required.`;
+  }
+
+  return errorModel;
+}
+
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  content: {
+    flex: 1,
+    padding: 16,
+    paddingBottom: 48,
+  },
+  feedbackModalIcon: {
+    height: 105,
+    width: 105
+  },
+  warningContainer: {
+    borderRadius: 8,
+    backgroundColor: 'hsl(47, 100%, 86%)', // warning yellow
+    marginBottom: 32,
+    borderWidth: 1,
+    borderColor: 'hsl(47, 100%, 70%)', // warning yellow - 16% light
+  },
+  warningMessage: {
+    fontSize: 14,
+    color: 'hsl(47, 100%, 22%)', // warning yellow - 64% light
+    padding: 12,
+  },
+  input: {
+    marginBottom: 24,
+  },
+  buttonContainer: {
+    alignSelf: 'stretch',
+    marginTop: 'auto',
+  },
+});
+
+export const CustomNetworkSettingsNav = Symbol('CustomNetworkSettings').toString();
+
+export const CustomNetworkSettingsScreen = ({ navigation }) => {
+  const dispatch = useDispatch();
+  const networkSettings = useSelector((state) => state.networkSettings);
+  const networkSettingsErrors = useSelector((state) => state.networkSettingsErrors);
+  const networkSettingsStatus = useSelector((state) => state.networkSettingsStatus);
+
+  const [formModel, setFormModel] = useState({
+    nodeUrl: networkSettings.nodeUrl,
+    explorerUrl: networkSettings.explorerUrl,
+    explorerServiceUrl: networkSettings.explorerServiceUrl,
+    walletServiceUrl: networkSettings.walletServiceUrl || '',
+    walletServiceWsUrl: networkSettings.walletServiceWsUrl || '',
+  });
+
+  const [errorModel, setErrorModel] = useState({
+    nodeUrl: networkSettingsErrors?.nodeUrl || '',
+    explorerUrl: networkSettingsErrors?.explorerUrl || '',
+    explorerServiceUrl: networkSettingsErrors?.explorerServiceUrl || '',
+    walletServiceUrl: networkSettingsErrors?.walletServiceUrl || '',
+    walletServiceWsUrl: networkSettingsErrors?.walletServiceWsUrl || '',
+  });
```

Comment:
Can you use something similar to this?

```
useState({...networkSettingsErrors})
```

Reply:
I can. But by using this verbose assignment, I take advantage of the type inference provided by the LSP. Without this declaration, it would be required to write the docstring anyway to take advantage of type inference.